source/speech/speech.py

```diff
@@ -125,6 +125,8 @@ class SpeechState:
 	oldRowSpan = None
 	oldColumnNumber = None
 	oldColumnSpan = None
+	lastReportedLanguage = None
+	suppressLanguageDescription = False
```

Comment:
I am quite concerned by this new state variable in the already complex `speak` function.

Have you investigated, or would you, the possibility to rework the code's architecture to make this more understandable?

If not possible, could you at least add a detailed description of the state variable and its usage?

Reply:
Thanks @CyrilleB79 for this new review. I'll bee adressing it in the next days, likely even this weekend to create an HTML document for testing.